### PR TITLE
Forward eth_sendTransaction to the sequencer

### PR DESCRIPTION
**Pull request by deffrian**
Fixes Closes Resolves #
In optimism we don't have local tx pool, so we need to forward all txs to the sequencer

## Changes

- Forward
- StepDependencyException.ThrowIfNull

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

#### Requires documentation update

- [ ] Yes
- [x] No

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Chains/base-mainnet.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
       "params": {
         "regolithTimestamp": "0x0",
         "bedrockBlockNumber": "0x0",
+        "sequencerUrl": "https://mainnet-sequencer.base.org",
```
Comment (jmederosalvarado) [inline src/Nethermind/Chains/base-mainnet.json:9]: I think this should be in config, not chainspec, chainspec should be for things affecting consensus

File: src/Nethermind/Chains/base-sepolia.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
         "l1FeeRecipient": "0x420000000000000000000000000000000000001A",
         "l1BlockAddress": "0x4200000000000000000000000000000000000015",
         "canyonBaseFeeChangeDenominator": "250",
+        "sequencerUrl": "https://sepolia-sequencer.base.org",
```
Comment (jmederosalvarado) [inline src/Nethermind/Chains/base-sepolia.json:13]: same as above, config, not chainspec

File: src/Nethermind/Chains/op-mainnet.json [hunk 3]
```diff
@@ -7,7 +7,8 @@
         "regolithTimestamp": "0x0",
         "bedrockBlockNumber": "0x645C277",
         "l1FeeRecipient": "0x420000000000000000000000000000000000001A",
-        "l1BlockAddress": "0x4200000000000000000000000000000000000015"
+        "l1BlockAddress": "0x4200000000000000000000000000000000000015",
+        "sequencerUrl": "https://mainnet-sequencer.optimism.io"
```
Comment (jmederosalvarado) [inline src/Nethermind/Chains/op-mainnet.json:11]: same as above, config, not chainspec

File: src/Nethermind/Chains/op-sepolia.json [hunk 4]
```diff
@@ -10,6 +10,7 @@
         "l1FeeRecipient": "0x420000000000000000000000000000000000001A",
         "l1BlockAddress": "0x4200000000000000000000000000000000000015",
         "canyonBaseFeeChangeDenominator": "250",
+        "sequencerUrl": "https://sepolia-sequencer.optimism.io",
```
Comment (jmederosalvarado) [inline src/Nethermind/Chains/op-sepolia.json:13]: same as above, config, not chainspec

File: src/Nethermind/Nethermind.Optimism/Rpc/OptimismEthRpcModule.cs [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+using Nethermind.JsonRpc.Client;
+using Nethermind.JsonRpc.Data;
+using Nethermind.JsonRpc.Modules.Eth;
+using Nethermind.Serialization.Rlp;
+using Nethermind.State.Proofs;
+using Nethermind.TxPool;
+
+namespace Nethermind.Optimism;
+
+public class OptimismEthRpcModule : IEthRpcModule
```
Comment (jmederosalvarado) [inline src/Nethermind/Nethermind.Optimism/Rpc/OptimismEthRpcModule.cs:25]: why not just inherit from `EthRpcModule` and override the `sendTransaction` methods?

Comment (deffrian) [reply src/Nethermind/Nethermind.Optimism/Rpc/OptimismEthRpcModule.cs:25]: We need to change Receipt type in getTxReceipt, so we need even different interface

**Review by jmederosalvarado — APPROVED** [review]

**Review by flcl42 — APPROVED** [review]

**Pull request merged**